File: src/api/prompt_templates.ts
import { invoke } from '@tauri-apps/api/core';
import type {
  PromptTemplate,
  CreatePromptTemplate,
  UpdatePromptTemplate,
} from '../types/ai';

export async function listPromptTemplates(): Promise<PromptTemplate[]> {
  return invoke<PromptTemplate[]>('list_prompt_templates');
}

export async function listPromptTemplatesByType(
  template_type: string
): Promise<PromptTemplate[]> {
  return invoke<PromptTemplate[]>('list_prompt_templates_by_type', { templateType: template_type });
}

export async function getPromptTemplate(id: string): Promise<PromptTemplate> {
  return invoke<PromptTemplate>('get_prompt_template', { id });
}

export async function createPromptTemplate(
  input: CreatePromptTemplate
): Promise<PromptTemplate> {
  return invoke<PromptTemplate>('create_prompt_template', { input });
}

export async function updatePromptTemplate(
  id: string,
  input: UpdatePromptTemplate
): Promise<PromptTemplate> {
  return invoke<PromptTemplate>('update_prompt_template', { id, input });
}

export async function deletePromptTemplate(id: string): Promise<void> {
  return invoke('delete_prompt_template', { id });
}

export async function resetPromptTemplateBuiltin(
  id: string
): Promise<PromptTemplate> {
  return invoke<PromptTemplate>('reset_prompt_template_builtin', { id });
}

File: src-tauri/src/commands/prompt_templates.rs
use sqlx::SqlitePool;
use tauri::State;

use crate::error::AppError;
use crate::models::prompt_template::{CreatePromptTemplate, PromptTemplate, UpdatePromptTemplate};
use crate::services::prompt_template_service;

#[tauri::command]
pub async fn list_prompt_templates(
    pool: State<'_, SqlitePool>,
) -> Result<Vec<PromptTemplate>, AppError> {
    prompt_template_service::list(pool.inner()).await
}

#[tauri::command]
pub async fn list_prompt_templates_by_type(
    pool: State<'_, SqlitePool>,
    template_type: String,
) -> Result<Vec<PromptTemplate>, AppError> {
    prompt_template_service::list_by_type(pool.inner(), &template_type).await
}

#[tauri::command]
pub async fn get_prompt_template(
    pool: State<'_, SqlitePool>,
    id: String,
) -> Result<PromptTemplate, AppError> {
    prompt_template_service::get(pool.inner(), &id).await
}

#[tauri::command]
pub async fn create_prompt_template(
    pool: State<'_, SqlitePool>,
    input: CreatePromptTemplate,
) -> Result<PromptTemplate, AppError> {
    prompt_template_service::create(pool.inner(), &input).await
}

#[tauri::command]
pub async fn update_prompt_template(
    pool: State<'_, SqlitePool>,
    id: String,
    input: UpdatePromptTemplate,
) -> Result<PromptTemplate, AppError> {
    prompt_template_service::update(pool.inner(), &id, &input).await
}

#[tauri::command]
pub async fn delete_prompt_template(
    pool: State<'_, SqlitePool>,
    id: String,
) -> Result<(), AppError> {
    prompt_template_service::delete(pool.inner(), &id).await
}

#[tauri::command]
pub async fn reset_prompt_template_builtin(
    pool: State<'_, SqlitePool>,
    id: String,
) -> Result<PromptTemplate, AppError> {
    prompt_template_service::reset_builtin(pool.inner(), &id).await
}
